components/apim-analytics/org.wso2.analytics.solutions.apim.analytics/widgets/APIMApiCreated/src/APIMApiCreated.jsx

```diff
@@ -0,0 +1,372 @@
+            },
+            weekCount: {
+                margin: 0,
+                marginTop: '5%',
+                color: 'rgb(135,205,223)',
+                letterSpacing: 1,
+                fontSize: '80%',
+            },
+            typeText: {
+                textAlign: 'left',
+                fontWeight: 'normal',
+                margin: 0,
+                display: 'inline',
+                marginLeft: '3%',
+                letterSpacing: 1.5,
+                fontSize: 'small',
+            },
+            icon: {
+                position: 'absolute',
+                bottom: '13%',
+                right: '8%',
+            },
+            loadingIcon: {
+                margin: 'auto',
+                display: 'block',
+            },
+        };
+
+        this.props.glContainer.on('resize', () => this.setState({
+            width: this.props.glContainer.width,
+            height: this.props.glContainer.height,
+        }));
+
+        this.assembleweekQuery = this.assembleweekQuery.bind(this);
+        this.assembletotalQuery = this.assembletotalQuery.bind(this);
+        this.handleWeekCountReceived = this.handleWeekCountReceived.bind(this);
+        this.handleTotalCountReceived = this.handleTotalCountReceived.bind(this);
+        this.loadLocale = this.loadLocale.bind(this);
+    }
+
+    componentDidMount() {
+        const locale = languageWithoutRegionCode || language;
+        this.loadLocale(locale).catch(() => {
+            this.loadLocale().catch(() => {
+                // TODO: Show error message.
+            });
+        });
+
+        super.getWidgetConfiguration(this.props.widgetID)
+            .then((message) => {
+                this.setState({
+                    providerConfig: message.data.configs.providerConfig,
+                }, this.assembletotalQuery);
+            })
+            .catch((error) => {
+                console.error("Error occurred when loading widget '" + this.props.widgetID + "'. " + error);
+                this.setState({
+                    faultyProviderConfig: true,
+                });
+            });
+    }
+
+    componentWillUnmount() {
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+    }
+
+    /**
+     * Load locale file.
+     * @param {string} locale Locale name
+     * @returns {Promise} Promise
+     * @memberof APIMApiCreated
+     */
+    loadLocale(locale = 'en') {
+        return new Promise((resolve, reject) => {
+            Axios.get(`${window.contextPath}/public/extensions/widgets/APIMApiCreated/locales/${locale}.json`)
+                .then((response) => {
+                    // eslint-disable-next-line global-require, import/no-dynamic-require
+                    addLocaleData(require(`react-intl/locale-data/${locale}`));
+                    this.setState({ localeMessages: defineMessages(response.data) });
+                    resolve();
+                })
+                .catch(error => reject(error));
```

Comment:
Lets add a `console.error` in Promise `catch` when there is no error message is shown to the user.